File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/editor/codestyle/java/javadoc/JavadocSettings.java
package com.github.alexandrecarlton.idea.settings.layout.editor.codestyle.java.javadoc;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@JsonDeserialize(as = ImmutableJavadocSettings.class)
public interface JavadocSettings {

  Optional<Boolean> enableJavadocFormatting();

  Optional<JavadocAlignmentSettings> alignment();

  Optional<JavadocBlankLinesSettings> blankLines();

  Optional<JavadocInvalidTagsSettings> invalidTags();

  Optional<JavadocOtherSettings> other();

}

File: src/main/kotlin/com/github/alexandrecarlton/idea/settings/layout/editor/codestyle/java/javadoc/JavadocSettings.kt
package com.github.alexandrecarlton.idea.settings.layout.editor.codestyle.java.javadoc

import com.fasterxml.jackson.annotation.JsonProperty

data class JavadocSettings(

    // TODO: This isn't used.
    @JsonProperty("Enable JavaDoc Formatting")
    val enableJavadocFormatting: Boolean? = null,

    @JsonProperty("Alignment")
    val alignment: JavadocAlignmentSettings? = null,

    @JsonProperty("Blank lines")
    val blankLines: JavadocBlankLinesSettings? = null,

    @JsonProperty("Invalid tags")
    val invalidTags: JavadocInvalidTagsSettings? = null,

    @JsonProperty("Other")
    val other: JavadocOtherSettings? = null
)
